games/hangman.c

```diff
@@ -0,0 +1,291 @@
+/**
+ * @file
+ * @brief C implementation of [Hangman Game](https://en.wikipedia.org/wiki/Hangman_(game))
+ * @details
+ * Simple, readable version of hangman.
+ * Changed graphic to duck instead of traditional stick figure (same number of guesses).
+ * @author [AtlantaEmrys2002](https://github.com/AtlantaEmrys2002)
+*/
+
+#include <ctype.h> /// for main() - tolower()
+#include <stdio.h> /// for main(), new_word(), new_guess(), won() - I/O operations
+#include <stdlib.h> /// for all functions - exit(), rand() and file functions
+#include <string.h> /// for main() - for string operations strlen, strchr, strcpy
+#include <time.h> /// for new_game() - used with srand() for declaring new game instance
+
+/*
+ * @brief game_instance structure that holds current state of game
+ */
+struct game_instance{
+
+    char current_word[30]; ///< word to be guessed by player
+    char hidden[30]; ///< hidden version of word that is displayed to player
+    int size; ///< size of word
+    int incorrect; ///< number of incorrect guesses
+    char guesses[25]; ///< previous guesses
+    int guesses_size; ///< size of guesses array
+
+};
+
+// function prototypes
+struct game_instance new_game(void); // creates a new game
+int new_guess(char, const char guesses[], int size); // checks if player has already played letter
+int in_word(char, const char word[], unsigned int size); // checks if letter is in word
+void picture(int score); // outputs image of duck (instead of hang man)
+void won(const char word[], int score); // checks if player has won or lost
+
+/**
+ * @brief Main Function
+ * @returns 0 on exit
+ */
+int main() {
+
+    struct game_instance game = new_game(); // new game created
+    char guess; // current letter guessed by player
+
+    // main loop - asks player for guesses
+    while ((strchr(game.hidden, '_') != NULL) && game.incorrect <= 12) {
```

Comment:
Calling `strchr()` every loop struck me as somewhat sad and inelegant, an alternative would be to catalog all the different letters in the word at new game creation, combining this with the four-byte bit array approach, we can potentially make it run very fast give how efficient bitwise operators are compared to iterating over array(s) and comparing manually. But yeah, I concede this is definitely too complex for a piece of code that is supposed to be readable and understandable